pkg/webhook/creator/validate_exec.go

```diff
@@ -0,0 +1,95 @@
+//
+// Copyright (c) 2019-2020 Red Hat, Inc.
+// This program and the accompanying materials are made
+// available under the terms of the Eclipse Public License 2.0
+// which is available at https://www.eclipse.org/legal/epl-2.0/
+//
+// SPDX-License-Identifier: EPL-2.0
+//
+// Contributors:
+//   Red Hat, Inc. - initial API and implementation
+//
+package creator
+
+import (
+	"context"
+	"fmt"
+	"github.com/che-incubator/che-workspace-operator/internal/controller"
+	"github.com/che-incubator/che-workspace-operator/pkg/controller/workspace/model"
+	"k8s.io/api/admission/v1beta1"
+	v1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/types"
+	"net/http"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/webhook/admission"
+)
+
+// WorkspaceExecValidator validates execs process all exec requests and:
+// if related pod DOES NOT have workspace_id label - just skip it
+// if related pod DOES have workspace_id label - make sure that exec is requested by workspace creator
+type WorkspaceExecValidator struct {
+	client  client.Client
+	decoder *admission.Decoder
+}
+
+func (v *WorkspaceExecValidator) Handle(ctx context.Context, req admission.Request) admission.Response {
+	if req.Resource.Resource == "pods" && req.Operation == v1beta1.Connect {
+		return v.handleCreateExec(ctx, req)
+	}
+	return admission.Denied(fmt.Sprintf("This admission is not supposed to handle %s operation for %s. Let administrator to know about this issue", req.Operation, req.Kind))
```

Comment:
When is this path triggered? Or is this just a sanity check?

Reply:
> Or is this just a sanity check?

I believe so ) Just to make sure that configuration and code work well together.